experiments/conductor/cmd/runner/controller_commands.go

```diff
@@ -0,0 +1,434 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runner
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// generateControllerClient creates a controller client for the branch
+// This implements the logic from 01-create-controller-client.sh
+func generateControllerClient(ctx context.Context, opts *RunnerOptions, branch Branch) ([]string, error) {
+	log.Printf("Generating controller client for branch %s", branch.Name)
+
+	// Check if we have the required fields
+	if branch.ProtoSvc == "" {
+		return nil, fmt.Errorf("branch %s is missing ProtoSvc field", branch.Name)
+	}
+
+	if branch.Group == "" {
+		return nil, fmt.Errorf("branch %s is missing Group field", branch.Name)
+	}
+
+	if branch.Kind == "" {
+		return nil, fmt.Errorf("branch %s is missing Kind field", branch.Name)
+	}
+
+	// Create the controller client
+	outputPath := filepath.Join("pkg", "controller", "direct", branch.Group, "client.go")
```

Comment:
`client.go` is not a typical file under controller. Some contributors manually add it, but it is not for generated code so far.

Reply:
i didnt know that. Since it is a separate command, users can choose not to run it for their branch. Or rather selectively run for some branches. 

contrary, Is it ok to create for all resources ?